### PR TITLE
api: add fields for propagation of ORCA to LRS

### PR DESCRIPTION
**Pull request by markdroth**
Commit Message: api: add fields for propagation of ORCA to LRS
Additional Description: Add fields in CDS to control which ORCA fields get propagated to LRS.  Also add fields in LRS for top-level ORCA metrics, to avoid having to encode the field name in LRS for these common cases.
Risk Level: Low
Testing: N/A
Docs Changes: Included in PR
Release Notes: N/A
Platform Specific Features: N/A

CC @efimki @andresguedez @ejona86 @dfawley @vishalpowar

### CODE REVIEW

**Comment by tonya11en** [comment]
format wants you to apply:
```
Changes made to following files:

  ./envoy/config/endpoint/v3/load_report.proto

Format check failed, try apply following patch to fix:
diff --git a/api/envoy/config/endpoint/v3/load_report.proto b/api/envoy/config/endpoint/v3/load_report.proto
index f91c9cb..5ad177a 100644
--- a/api/envoy/config/endpoint/v3/load_report.proto
+++ b/api/envoy/config/endpoint/v3/load_report.proto
@@ -78,8 +78,11 @@ message UpstreamLocalityStats {
   // Stats for multi-dimensional load balancing.
   // These typically come from endpoint metrics reported via ORCA.
   UnnamedEndpointLoadMetricStats cpu_utilization = 12;
+
   UnnamedEndpointLoadMetricStats mem_utilization = 13;
+
   UnnamedEndpointLoadMetricStats application_utilization = 14;
+
   repeated EndpointLoadMetricStats load_metric_stats = 5;
 
   // Endpoint granularity stats information for this locality. This information

```

**Review by mattklein123 — APPROVED** [review]

**Review by adisuissa — APPROVED** [review]
Thanks!

/lgtm api

**Review by adisuissa — APPROVED** [review]
Thanks!

/lgtm api

File: api/envoy/config/cluster/v3/cluster.proto [hunk 1]
```diff
@@ -1151,6 +1151,22 @@ message Cluster {
...
+  // ``named_metrics`` field.
+  //
+  // The special map key ``*`` means to report all entries in the map (e.g., ``named_metrics.*`` means to
+  // report all entries in the ORCA named_metrics field). Note that this should be used only with trusted
+  // backends.
+  //
+  // The metric names in LRS will follow the same semantics as this field. In other words, if this field
+  // contains ``named_metrics.foo``, then the LRS load report will include the data with that same string
+  // as the key.
+  repeated string lrs_report_endpoint_metric = 57;
```
Comment (dfawley) [inline api/envoy/config/cluster/v3/cluster.proto:1168]: Note that this isn't a plural, but all the other repeated fields I'm seeing adjacent to this PR are plural.  Not sure if this kind of consistency is important to Envoy maintainers, just calling it out.

Comment (wbpcode) [reply api/envoy/config/cluster/v3/cluster.proto:1168]: Hi, @markdroth is this field used in gRPC or not? If it haven't be released in gRPC, I think we can re-evaluate/update it.

Comment (markdroth) [reply api/envoy/config/cluster/v3/cluster.proto:1168]: We haven't started using it yet.  I'm fine with changing the name to be plural.

Comment (markdroth) [reply api/envoy/config/cluster/v3/cluster.proto:1168]: I sent #35284 to make this change.

**Pull request merged**